bmwqemu.pm

```diff
@@ -203,27 +203,50 @@ sub pp (@args) {
     return $value_with_trailing_newline;
 }
 
+# Use special argument `-masked` to hide the given value in log output.
+# It can be specified multiple times and or the value can be a ARRAY_REF or
+# scalar.
 sub log_call (@args) {
     my $fname = (caller(1))[3];
     update_line_number();
+
+    # extract -masked parameter out of argument list
+    my $masked = [];
+    my @tmpargs;
+    while (@args) {
+        my $v = shift @args;
+        if (defined($v) && $v eq '-masked' && @args) {
+            my $mval = shift @args;
+            $masked = [@$masked, ref($mval) eq 'ARRAY' ? @$mval : $mval];
```

Comment:
```suggestion
            push @$masked, ref($mval) eq 'ARRAY' ? @$mval : $mval;
```

You could also make `$masked` an array (instead of an array ref) which would be more consistent with other local variables.